fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/InstrumentedTransaction.java

```diff
@@ -0,0 +1,378 @@
+        underlying.mutate(opType, key, param);
+        /* Do we want to track each mutation type separately as well? */
+        timer.increment(FDBStoreTimer.Counts.MUTATIONS);
+    }
+
+    @Override
+    public CompletableFuture<Void> commit() {
+        long startTimeNanos = System.nanoTime();
+        return underlying.commit().whenComplete((v, ex) ->
+                timer.recordSinceNanoTime(FDBStoreTimer.Events.COMMITS, startTimeNanos));
+    }
+
+    @Override
+    public Long getCommittedVersion() {
+        return underlying.getCommittedVersion();
+    }
+
+    @Override
+    public CompletableFuture<byte[]> getVersionstamp() {
+        return underlying.getVersionstamp();
+    }
+
+    @Override
+    public CompletableFuture<Transaction> onError(Throwable throwable) {
+        return underlying.onError(throwable);
+    }
+
+    @Override
+    public void cancel() {
+        underlying.cancel();
+    }
+
+    @Override
+    public CompletableFuture<Void> watch(byte[] bytes) throws FDBException {
+        return underlying.watch(bytes);
+    }
+
+    @Override
+    public Database getDatabase() {
+        return underlying.getDatabase();
+    }
+
+    @Override
+    public <T> T run(Function<? super Transaction, T> function) {
+        return function.apply(this);
+    }
+
+    @Override
+    public <T> CompletableFuture<T> runAsync(Function<? super Transaction, ? extends CompletableFuture<T>> function) {
+        return AsyncUtil.applySafely(function, this);
+    }
+
+    @Override
+    public void close() {
+        underlying.close();
+    }
+
+    @Override
+    public boolean isSnapshot() {
+        return underlying.isSnapshot();
+    }
+
+    @Override
+    public ReadTransaction snapshot() {
+        return underlying.snapshot();
+    }
+
+    @Override
+    public CompletableFuture<Long> getReadVersion() {
+        return underlying.getReadVersion();
+    }
+
+    @Override
+    public void setReadVersion(long l) {
+        underlying.setReadVersion(l);
+    }
+
+    @Override
+    public boolean addReadConflictRangeIfNotSnapshot(byte[] keyBegin, byte[] keyEnd) {
+        return underlying.addReadConflictRangeIfNotSnapshot(keyBegin, keyEnd);
+    }
+
+    @Override
+    public boolean addReadConflictKeyIfNotSnapshot(byte[] key) {
+        return underlying.addReadConflictKeyIfNotSnapshot(key);
+    }
+
+    @Override
+    public CompletableFuture<byte[]> get(byte[] key) {
+        timer.increment(FDBStoreTimer.Counts.READS);
+        return underlying.get(key).thenApply(this::recordRead);
+    }
+
+    @Override
+    public CompletableFuture<byte[]> getKey(KeySelector keySelector) {
+        timer.increment(FDBStoreTimer.Counts.READS);
+        return underlying.getKey(keySelector).thenApply(this::recordRead);
+    }
+
+    @Override
+    public AsyncIterable<KeyValue> getRange(KeySelector begin, KeySelector end) {
+        /* Should this could as one read? */
+        timer.increment(FDBStoreTimer.Counts.READS);
```

Comment:
As noted, for these iterators, they are being counted a single read. Is this ok?  Can we do better?